submissions/Ant-C-tech/oop-exercise/index.js

```diff
@@ -0,0 +1,113 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(species, name, legs, saying, friends) {
+        this.species = species;
+        this.name = name;
+        this.legs = legs;
+        this.friends = friends;
+        this.saying = saying;
+
+        this.templete = [`I am a ${this.species}`,
+            `My name is ${ this.name }`,
+            `I have ${this.legs} legs`,
+            `I want to say you: "${this.saying}"`
+        ].join('. ');
+    };
+
+    getVoice() {
+        return (this.friends) ? [
+            this.templete,
+            `I have friend${(this.friends.length > 1) ? 's' : ''}: ${this.friends.join(', ')}`
+        ].join('. ') : [
+            this.templete,
+            'I am looking for friends'
+        ].join('. ');
```

Comment:
This fragment, simplified, looks as:
```
return condition
  ? [variableA, stringifiedFriends].join()
  : [variableA, 'string literal'].join()
```

In fact `condition` manages one of the resulting arrays elements. So just move it there. This way scalability and maintainability get improved, and the code gets shorter as quite a number of repetitions get eliminated.

Example above refactored:
```
return [variableA, 
  condition ? stringifiedFriends : 'string literal'].join()
```
or added formatting for even better readability:
```
return [
  variableA, 
  condition
    ? stringifiedFriends 
    : 'string literal'
  ].join()
```

Reply:
Alternatively you might have had
```
const stringifiedFriends = ... all the magic with friends here...;
return [variableA, stringifiedFriends].join();
```

Ternaries inside template strings make reading painful